From 0758e69eb0622f0eeadd5bae12981d735bc62c8e Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Thu, 7 Mar 2024 11:06:20 +0100
Subject: [PATCH 0904/1141] arm64: dts: st: remove USB33 from stm32mp25

Remove PWR USB33 that has been suppressed from the ref manual.
USB vdd33-supply is directly supplied by the PMIC LDO4 via
scmi_vdd3v3_usb.

Change-Id: I48bbe370fcc0aadf7929a42b975277a9ce8ab51a
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/365099
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi     | 7 -------
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts  | 6 ++++++
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 6 ++++++
 3 files changed, 12 insertions(+), 7 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 392ab416466e..68c8fe54dbb1 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -153,11 +153,6 @@ scmi_vdd33ucpd: regulator@5 {
 						reg = <VOLTD_SCMI_UCPD>;
 						regulator-name = "vdd33ucpd";
 					};
-					scmi_vdd33usb: regulator@6 {
-						voltd-name = "vdd33usb";
-						reg = <VOLTD_SCMI_USB33>;
-						regulator-name = "vdd33usb";
-					};
 					scmi_vdda18adc: regulator@7 {
 						voltd-name = "vdda18adc";
 						reg = <VOLTD_SCMI_ADC>;
@@ -261,7 +256,6 @@ usb2_phy1: usb2-phy1 {
 		st,syscfg       = <&syscfg 0x2400>;
 		clocks          = <&rcc CK_KER_USB2PHY1>;
 		resets          = <&rcc USB2PHY1_R>;
-		vdd33-supply    = <&scmi_vdd33usb>;
 		status          = "disabled";
 	};
 
@@ -271,7 +265,6 @@ usb2_phy2: usb2-phy2 {
 		st,syscfg       = <&syscfg 0x2800>;
 		clocks          = <&rcc CK_KER_USB2PHY2EN>;
 		resets          = <&rcc USB2PHY2_R>;
-		vdd33-supply    = <&scmi_vdd33usb>;
 		status          = "disabled";
 	};
 
diff --git a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
index 1982721cba2a..f5d56eeeba55 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -640,6 +640,10 @@ scmi_vdd_emmc: regulator@18 {
 		reg = <VOLTD_SCMI_STPMIC2_LDO2>;
 		regulator-name = "vdd_emmc";
 	};
+	scmi_vdd3v3_usb: regulator@20 {
+		reg = <VOLTD_SCMI_STPMIC2_LDO4>;
+		regulator-name = "vdd3v3_usb";
+	};
 	scmi_v5v_hdmi: regulator@21 {
 		reg = <VOLTD_SCMI_STPMIC2_LDO5>;
 		regulator-name = "v5v_hdmi";
@@ -750,10 +754,12 @@ &usart6 {
 };
 
 &usb2_phy1 {
+	vdd33-supply = <&scmi_vdd3v3_usb>;
 	status = "okay";
 };
 
 &usb2_phy2 {
+	vdd33-supply = <&scmi_vdd3v3_usb>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
index 9757598f4d85..f950cabbd5dd 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -580,6 +580,10 @@ scmi_vdd_emmc: regulator@18 {
 		reg = <VOLTD_SCMI_STPMIC2_LDO2>;
 		regulator-name = "vdd_emmc";
 	};
+	scmi_vdd3v3_usb: regulator@20 {
+		reg = <VOLTD_SCMI_STPMIC2_LDO4>;
+		regulator-name = "vdd3v3_usb";
+	};
 	scmi_vdd_sdcard: regulator@23 {
 		reg = <VOLTD_SCMI_STPMIC2_LDO7>;
 		regulator-name = "vdd_sdcard";
@@ -678,10 +682,12 @@ &usart6 {
 };
 
 &usb2_phy1 {
+	vdd33-supply = <&scmi_vdd3v3_usb>;
 	status = "okay";
 };
 
 &usb2_phy2 {
+	vdd33-supply = <&scmi_vdd3v3_usb>;
 	status = "okay";
 };
 
-- 
2.39.2

